sw/device/silicon_creator/lib/ownership/ecdsa.c

```diff
@@ -0,0 +1,50 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/silicon_creator/lib/ownership/ecdsa.h"
+
+#include <stdbool.h>
+
+#include "sw/device/lib/base/macros.h"
+#include "sw/device/silicon_creator/lib/drivers/hmac.h"
+#ifdef USE_CRYPTOC
+// TODO(cfrantz): Replace the CryptoC implementation with a native OpenTitan
+// implementation.
+#include "sw/vendor/cryptoc/include/cryptoc/p256.h"
+#include "sw/vendor/cryptoc/include/cryptoc/p256_ecdsa.h"
+
+// This satisfies cryptoc's use of the assert macro.
+OT_WEAK void __assert_func(const char *file, int line, const char *func,
+                           const char *expr) {
+  while (true) {
+    HARDENED_TRAP();
+  }
+}
+#else
+#error "No ECDSA implementation for lib/ownership."
+#endif
+
+hardened_bool_t ecdsa_verify_digest(const owner_key_t *pubkey,
```

Comment:
Just an FYI: (in the future) we may be able to wire this up to the ECDSA sigverify that is loaded in the single OTBN [boot services lib](https://cs.opensource.google/opentitan/opentitan/+/master:sw/device/silicon_creator/lib/otbn_boot_services.h;drc=6a3bec2f36b7e7bb4bc70f2e8f03c31640e4cd6d;l=134), which is backed by the OTBN program that will already be loaded by the ROM.

Reply:
That is the plan and the reason these cut-outs exist for now.